### PR TITLE
Minor fix to MultiappVariableValueSampleTransfer, minor FV asserts fix

### LINKED ISSUE #17194: MultiAppVariableValueSample doesn't update the subdomain id of its assembly
## Bug Description
Hitting an assertion because the element subdomain id does not match the assembly current elem subdomain id

## Steps to Reproduce
Use a MultiAppVariableValueSample with solve=false

## Impact
Have to disable the assert to debug something else. Fix is in a PR coming soon

### PR DESCRIPTION
**Pull request by GiudGiud**
Closes #17194 #17193 

Any ideas to reduce the number of warnings output in ComputeFVFluxThread? Maybe a boolean?

### CODE REVIEW

**Review by loganharbour — DISMISSED** [review]
Test? 

I also believe this should be an error, not a warning, no?

**Comment by loganharbour** [comment]
> Any ideas to reduce the number of warnings output in ComputeFVFluxThread? Maybe a boolean?

If you're settled on it being a warning, use `mooseDoOnce(mooseWarning(...))`

File: framework/include/loops/ComputeFVFluxThread.h [hunk 1]
```diff
@@ -489,25 +489,35 @@ ComputeFVFluxThread<RangeType>::checkPropDeps(
...
   }
 
+  // Print a warning if block restricted materials are used
+  std::stringstream same_matprop_name_str;
+  std::copy(same_matprop_name.begin(),
+            same_matprop_name.end(),
+            std::ostream_iterator<int>(same_matprop_name_str, " "));
+
+  if (same_matprop_name.size() > 0)
+    mooseWarning("Multiple objects supply properties of IDs",
```
Comment (loganharbour) [inline framework/include/loops/ComputeFVFluxThread.h:513]: After looking at the test - you should really add some `\n` up in here so this isn't an insane thing that appears across your entire screen and the next screen

**Review by loganharbour — CHANGES_REQUESTED** [review]

File: framework/include/loops/ComputeFVFluxThread.h (hunk 1 above)
Comment (GiudGiud) [reply framework/include/loops/ComputeFVFluxThread.h]: done

**Comment by GiudGiud** [comment]
It would be an error if it was rigorously checking for the problematic situation to exist. But it's just checking for the situation to likely exist, so it should be just a warning. 

If you look at the code, it's just checking for block restricted materials with the same name. It is not cross checking with the kernels to see if they are also involved in block restricted kernels.

**Comment by GiudGiud** [comment]
@loganharbour is the modified test a valid way to check for warnings ? (RunException without allowing warnings)

**Review by loganharbour — APPROVED** [review]

**Comment by loganharbour** [comment]
What you have there is mostly valid. You can add `allow_warnings = true`  in the test spec and that's fine too

**Comment by lindsayad** [comment]
I'd like to be ping'd when we're changing assertions that I put in. I think I'm ok with this, but would like to think on it first next time if possible

**Pull request merged**